download/client.go

```diff
@@ -34,14 +36,18 @@ type metadata struct {
 	Versions []string `xml:"versioning>versions>version"`
 }
 
-func GetVersions(groupId, artifactId, repository, username, password string) ([]string, error) {
+type Downloader interface {
+	GetVersions(repository string) ([]string, error)
```

Comment:
i think you'll need groupId & artifactId to be able to construct the url to get to the meta-data.xml